pkg/endpoints/pipeline_test.go

```diff
@@ -1145,3 +1085,245 @@ func TestPipelineRunUpdateStatusAlreadySet412(t *testing.T) {
 		t.Errorf("FAIL: should have received a http 412 code when setting the status to something already set, got %d", resp.StatusCode())
 	}
 }
+
+/* PipelineRun create bad request 400 */
+func TestCreatePipelineRunBadRequest(t *testing.T) {
+
+	r := dummyResource()
+
+	pipeline1 := v1alpha1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "Pipeline1",
+		},
+		Spec: v1alpha1.PipelineSpec{},
+	}
+
+	_, err := r.PipelineClient.TektonV1alpha1().Pipelines("ns1").Create(&pipeline1)
+	if err != nil {
+		t.Errorf("FAIL: error creating test pipeline: %s", err)
+	}
+
+	httpWriter := httptest.NewRecorder()
+
+	badRequestPipelineRunBody := strings.NewReader(`{"fielddoesnotexist": "pipeline",}`)
+
+	badRequestPipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", badRequestPipelineRunBody)
+	badRequestPipelineRunRequestRestful := dummyRestfulRequest(badRequestPipelineRunRequest, "ns1", "")
+	resp := dummyRestfulResponse(httpWriter)
+	r.createPipelineRun(badRequestPipelineRunRequestRestful, resp)
```

Comment:
I'm seeing these lines repeated over and over again in your tests:
```
httpWriter := httptest.NewRecorder()

pipelineRunBody := strings.NewReader(`TEXT`)

pipelineRunRequest := dummyHTTPRequest("POST", "http://wwww.dummy.com:8383/v1/namespaces/ns1/pipelinerun", pipelineRunBody)
pipelineRunRequestRestful := dummyRestfulRequest(pipelineRunRequest, "ns1", "")
resp := dummyRestfulResponse(httpWriter)
r.createPipelineRun(pipelineRunRequestRestful, resp)
```
Please make a function for the repeated code to cut down own code bloat and make the tests easier to read.